pkg/components/rook/component.go

```diff
@@ -116,8 +117,20 @@ func (c *component) Metadata() components.Metadata {
 func convertNodeSelector(m map[string]string) string {
 	var ret string
 
-	for k, v := range m {
-		ret += fmt.Sprintf("%s=%s; ", k, v)
+	// covert map to slice and sort them
+	// by keys.
+	keys := make([]string, len(m))
+	i := 0
+
+	for k := range m {
+		keys[i] = k
+		i++
+	}
+
+	sort.Strings(keys)
+
+	for _, k := range keys {
+		ret += fmt.Sprintf("%s=%s; ", k, m[k])
```

Comment:
Not related to this PR, but it would perhaps be nicer to format to slice and then just return `strings.Join()`.